app/Http/Controllers/Admin/ModulesController.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace App\Http\Controllers\Admin;
+
+use App\Contracts\Controller;
+use App\Models\Module;
+use App\Services\ModuleService;
+use Illuminate\Filesystem\Filesystem;
+use Illuminate\Http\Request;
+use Laracasts\Flash\Flash;
+
+class ModulesController extends Controller
+{
+    private $moduleSvc;
+
+    protected $paths = [];
+    /**
+     * @var Filesystem
+     */
+    private $files;
+
+    public function __construct(ModuleService $moduleSvc, Filesystem $files)
+    {
+        $this->moduleSvc = $moduleSvc;
+        $this->files = $files;
+    }
+
+    /**
+     * Display a listing of the resource.
+     *
+     * @return mixed
+     */
+    public function index()
+    {
+        $modules = Module::all()->sortByDesc('id');
```

Comment:
The scan should be called here, so `$new_modules = $this->module->scan()`, and that should be passed to the template so it can render it as a new section

Reply:
Alright! Here `$this->module` would be instance of `App\Models\Module` ?

Reply:
Sorry - `$this->moduleSvc->scan()`